cmake/Modules/G4BuildSettings.cmake

```diff
@@ -59,6 +59,10 @@ include(CheckCXXSourceCompiles)
 include(IntelCompileFeatures)
 include(MSVCCompileFeatures)
 
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
This should be made into a cache variable _not_ hardcoded. For back compatibility, it should be set to `GEANT4_BUILD_CXXSTD` if that was defined.

The decision to default to C++17 has not been made yet.

Reply:
Uhm, but why can't it be hardcoded? If I understand correctly, CMake falls back to the highest available standard that the compiler support (unless CMAKE_CXX_REQUIRE_STANDARD is on).

I mean, setting this should not cause the build to fail, even on old platforms. What would cause it to fail would be using C++17/20 features.